### PR TITLE
Account verification

### PR DESCRIPTION
**Pull request by VelislavaC4Nexus**
### All pull requests:
* [ ] Have you considered security (e.g., XSS)?
* [ ] Have you considered desktop, mobile, and tablet form-factors?
* [ ] Have you considered [accessibility best practices](https://www.w3.org/WAI/standards-guidelines/wcag/)?

### Code
* [ ] Are the commits squashed into one commit?
* [ ] Is the [branch name and commit message](CONTRIBUTING.md#conventions-for-branch-names-and-commit-messages) following team convention?
* [ ] Is the code linted?
* [ ] Are all open issues, questions, and concerns by reviewers answered, resolved, and reviewed?

### Quality assurance
* [ ] If applicable, are unit tests written and are they passing?
* [ ] If applicable, are integration tests written and are they passing?
* [ ] Have checks linked in your pull request passed?

### Documentation
* [ ] Are there meaningful code comments included?

## Applicable to community contributors:
* [ ] Have you granted SFRA team access to your fork? [Grant access](CONTRIBUTING.md#community-contributors) 

### Documentation
* [ ] If applicable, is the change log updated?
* [ ] If applicable, have any UI text changes been reviewed by Documentation?
* [ ] If applicable, have any UI implementations been reviewed by the UX team?

### Security
* [ ] If applicable, has Security reviewed this code?
* [ ] If applicable, is a 3PP request submitted?

### CODE REVIEW

File: cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var base = require('base/login/login')
+
+var formValidation = require('base/components/formValidation');
+var createErrorNotification = require('base/components/errorNotification');
+
+base.register = function () {
+    $('form.registration').submit(function (e) {
+        console.log('register-cusrtom');
```
Comment (kalipiev) [inline cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js]: Please remove console.log

File: cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js
Comment (kalipiev) [inline cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js:1]: Some general guidelines for this file:
- remove console.logs
- When accessing elements from the DOM, please use CSS classes with js- prefix like "js-some-class", those will be used only by the JS.

File: cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+                form.spinner().stop();
+            }
+        });
+        return false;
+    });
+}
+
+module.exports = base
+
```
Comment (kalipiev) [inline cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js]: unnecessary empty lines

File: cartridges/app_custom_account_verification/cartridge/controllers/Account.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+ * @param {serverfunction} - get
+ */
+server.get('Verify', function (req, res, next) {
+    var CustomObjectMgr = require("dw/object/CustomObjectMgr");
+    var URLUtils = require("dw/web/URLUtils");
+    var accountVerifyCO = 'ACCOUNT_VERIFY_CO'
+
+    var accountId = req.querystring.accountId;
+    var email = req.querystring.email;
+    var existingCustomer = CustomerMgr.getCustomerByLogin(email);
```
Comment (kalipiev) [inline cartridges/app_custom_account_verification/cartridge/controllers/Account.js]: Please use helper function for the logic below

File: cartridges/app_custom_account_verification/cartridge/controllers/Login.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ * @param {serverfunction} - get
+ */
+server.append(
+    'Show',
+    consentTracking.consent,
+    server.middleware.https,
+    csrfProtection.generateToken,
+    function (req, res, next) {
+        var viewData = res.getViewData()
+        viewData.accountCustomObject = req.querystring.accountCustomObject
```
Comment (kalipiev) [inline cartridges/app_custom_account_verification/cartridge/controllers/Login.js]: Why do we need this ?

File: cartridges/app_custom_account_verification/cartridge/static/default/js/login.js
Comment (kalipiev) [inline cartridges/app_custom_account_verification/cartridge/static/default/js/login.js:1]: do not commit static files

File: cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js (hunk 1 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js]: fixed

File: cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js:1]: fixed

File: cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js (hunk 2 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_verification/cartridge/client/default/js/login/login.js]: removed

File: cartridges/app_custom_account_verification/cartridge/controllers/Account.js (hunk 3 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_verification/cartridge/controllers/Account.js]: moved logic into helper function

File: cartridges/app_custom_account_verification/cartridge/controllers/Login.js (hunk 4 above)
Comment (VelislavaC4Nexus) [reply cartridges/app_custom_account_verification/cartridge/controllers/Login.js]: I fixed it as i use "status". I need this to extend view data to display message if user has already verified or the verification link has expired

**Pull request merged**